devito/finite_differences/derivative.py

```diff
@@ -149,33 +150,58 @@ def __new__(cls, expr, *dims, **kwargs):
             else:
                 dcounter[d] += o
 
-        # Default finite difference orders depending on input dimension (.dt or .dx)
         # It's possible that the expr is a `sympy.Number` at this point, which
         # has derivative 0, unless we're taking a 0th derivative.
         if isinstance(expr, sympy.Number):
             if any(dcounter.values()):
                 return 0
             else:
                 return expr
+
+        # Use `fd_order` if specified
+        fd_order = kwargs.get('fd_order')
+        if fd_order is not None:
+            # If `fd_order` is specified collect these together
+            fcounter = defaultdict(int)
+            for d, o in zip(dims, as_tuple(fd_order)):
```

Comment:
Nitpick: Some extra comments on this big stack of loops and conditionals would help, since my eyes are glazing over looking at it